chia/full_node/full_node.py

```diff
@@ -389,6 +390,13 @@ async def manage(self) -> AsyncIterator[None]:
                         with contextlib.suppress(asyncio.CancelledError):
                             self.log.info(f"Awaiting long sync task {one_sync_task.get_name()}")
                             await one_sync_task
+                for segment_task in self._segment_task_list:
+                    if segment_task.done():
+                        self.log.info(f"segment task {segment_task.get_name()} done")
+                    else:
+                        with contextlib.suppress(asyncio.CancelledError):
+                            self.log.info(f"Awaitin segment task {segment_task.get_name()}")
+                            await segment_task
```

Comment:
I would think `await` on a task that's done would be a no-op, so you could probably just `await` unconditionally

Reply:
fyi, there's https://github.com/Chia-Network/chia-blockchain/pull/18914 waiting for arvid's review.  it would handle holding a reference to the task.  it is _not_ intended as a permanent solution but rather an easy way to make all our tasks have references without duplicating the tracking lists everywhere in slightly different ways each place.  it is just a stop gap until we get to the point we can use proper tooling (task groups) to manage our tasks.

anyways, if you like i think that could replace some aspects of this pr.  one of the basic things that's really hard (impossible?) to get right here is the consumption of the cancellation error.  you know you might get one from the task because you cancelled it and you don't want to propagate that, but you can't tell if the cancellation you get is from that task being cancelled or from a separate cancellation of the await right here...